src/System.Linq.Parallel/src/System/Linq/ParallelEnumerable.cs

```diff
@@ -4414,26 +4421,47 @@ public static bool SequenceEqual<TSource>(this ParallelQuery<TSource> first, Par
 #endif
                 catch (Exception ex)
                 {
+                    if ((firstThrew && firstBare) || (secondThrew && secondBare) && ex is AggregateException)
+                    {
+                        throw ex;
+                    }
                     ExceptionAggregator.ThrowOCEorAggregateException(ex, settings.CancellationState);
                 }
                 finally
                 {
-                    DisposeEnumerator<TSource>(e2, settings.CancellationState);
+                    DisposeEnumerator<TSource>(e2, settings.CancellationState, secondBare);
                 }
             }
             finally
             {
-                DisposeEnumerator<TSource>(e1, settings.CancellationState);
+                DisposeEnumerator<TSource>(e1, settings.CancellationState, firstBare);
             }
 
             return true;
         }
 
         /// <summary>
+        /// Move the enumerator to the next element, setting a flag if it threw an exception.
+        /// </summary>
+        private static bool MoveNextOrMark<TSource>(IEnumerator<TSource> e, ref bool threw)
```

Comment:
Haven't investigated.  I only added this because of how `SequenceEqual` double-nests exceptions, unlike every other operator (ie, to leave exception behavior unchanged).  I'd prefer to remove it, and have it only do the single nesting.